### PR TITLE
[Perf] Optimize popper component by not triggering layout on every render

### LINKED ISSUE #4837: [Perf] Optimize Popper component
Popper creates a new instance of Popper from popper.js on every render and this inturn triggers a layout re calculation which takes 10-20ms. 

PopperOptimization-Before

### PR DESCRIPTION
**Pull request by SatishGandham**
## Description
Poppers useEffect is called on every render, this triggers a layout calculation which takes 10-20 seconds. With this PR we only do that when required. Which is mostly the first render.

Fixes #4837  

## Type of change

- Performance improvement

## How Has This Been Tested?

Profiling before and after

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

PopperOptimization-Before
PopperOptimization-After

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: perf/ui/optimize-popper-component/4837 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | app/client/src/pages/Editor/Popper.tsx | 85.37 **(-0.34)** | 71.74 **(0)** | 88.89 **(0)** | 85.37 **(-0.34)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/Popper.tsx [hunk 1]
```diff
@@ -60,9 +59,13 @@ export default (props: PopperProps) => {
     onPositionChange = noop,
     themeMode = props.themeMode || ThemeMode.LIGHT,
   } = props;
-  const popperTheme = useSelector((state: AppState) =>
-    getThemeDetails(state, themeMode),
+  // Meomoizing to avoid rerender of draggable icon.
+  // What is the cost of memoizing?
+  const popperTheme = useMemo(
```
Comment (riodeuno) [inline app/client/src/pages/Editor/Popper.tsx:64]: Answering your query in the comments.
As long as `themeMode` is dynamic and can change across component updates, we should not worry about the cost of memoizing. The memoization also has a cache size of 1, so as long as `themeMode` is the same primitive value (string in this case), this should not re-calculate.

**Review by riodeuno — APPROVED** [review]

File: app/client/src/pages/Editor/Popper.tsx (hunk 1 above)
Comment (SatishGandham) [reply app/client/src/pages/Editor/Popper.tsx:64]: Without memoizing this, we sonly rerender a simple styled component, so was not sure if the cost of memoizing it was worth. I did not know the cache size is 1.

**Review by riodeuno — APPROVED** [review]

**Review by riodeuno — APPROVED** [review]

**Review by riodeuno — APPROVED** [review]

**Pull request merged**